test/test_evaluators.py

```diff
@@ -7,10 +7,75 @@
 from vision_evaluation.evaluators import AveragePrecisionEvaluator, F1ScoreEvaluator, TopKAccuracyEvaluator, ThresholdAccuracyEvaluator, MeanAveragePrecisionEvaluatorForSingleIOU, EceLossEvaluator, \
     PrecisionEvaluator, RecallEvaluator, TagWiseAccuracyEvaluator, TagWiseAveragePrecisionEvaluator, MeanAveragePrecisionNPointsEvaluator, BalancedAccuracyScoreEvaluator, \
     CocoMeanAveragePrecisionEvaluator, BleuScoreEvaluator, METEORScoreEvaluator, ROUGELScoreEvaluator, CIDErScoreEvaluator, SPICEScoreEvaluator, RocAucEvaluator, MeanIOUEvaluator, \
-    ForegroundIOUEvaluator, BoundaryMeanIOUEvaluator, BoundaryForegroundIOUEvaluator, L1ErrorEvaluator
+    ForegroundIOUEvaluator, BoundaryMeanIOUEvaluator, BoundaryForegroundIOUEvaluator, L1ErrorEvaluator, GroupWiseEvaluator
 from vision_evaluation.prediction_filters import TopKPredictionFilter, ThresholdPredictionFilter
 
 
+class TestGroupWiseClassificationEvaluator(unittest.TestCase):
+    TARGETS = [np.array([[0, 0],
+                        [0, 0],
+                        [0, 1],
+                        [0, 1],
+                        [2, 0],
+                        [2, 0]]),
+               np.array([[0, 0],
+                        [1, 0],
+                        [0, 1],
+                        [0, 1],
+                        [1, 0],
+                        [1, 0]])]
+    PREDICTIONS = [np.array([[0.8, 0.1, 0.1],
+                            [0.8, 0.1, 0.1],
+                            [0.1, 0.8, 0.1],
+                            [0.1, 0.8, 0.1],
+                            [0.1, 0.1, 0.8],
+                            [0.1, 0.1, 0.8]]),
+                   np.array([[0.8, 0.1],
+                            [0.1, 0.8],
+                            [0.8, 0.1],
+                            [0.1, 0.8],
+                            [0.8, 0.1],
+                            [0.8, 0.1]])]
+
+    def test_group_wise_evaluator(self):
+        gts = [[{(0, 0): 1,
+                (0, 1): 0,
+                (2, 0): 1},
+               {(0, 0): 1,
+                (0, 1): 1,
+                (2, 0): 1}],
+
+               [{(0, 0): 1,
+                (0, 1): 0.5,
+                (1, 0): 0.333},
+               {(0, 0): 1,
+                (0, 1): 1,
+                (1, 0): 1}]]
+
+        for k_idx, top_k in enumerate([1, 5]):
+
+            for i, (targets, predictions) in enumerate(zip(self.TARGETS, self.PREDICTIONS)):
+
+                eval_fn = TopKAccuracyEvaluator(top_k)
```

Comment:
can you add a test for group-wise mAP calculation for detection case as well? The corresponding evaluator is CocoMeanAveragePrecisionEvaluator

Reply:
updated.